cfme/tests/openstack/cloud/test_flavors.py

```diff
@@ -0,0 +1,142 @@
+def flavor(appliance, provider):
+    flavor = appliance.collections.cloud_flavors.create(name=fauxfactory.gen_alpha(),
+                               provider=provider,
+                               ram=RAM,
+                               vcpus=VCPUS,
+                               disk=DISK_SIZE,
+                               swap=SWAP_SIZE,
+                               rxtx=RXTX)
+    yield flavor
+
+    if flavor.exists:
+        flavor.delete(wait=False)
+
+
+@pytest.fixture(scope='function')
+def zero_disk_flavor(provider):
+    api_flv = provider.mgmt.api.flavors.create(fauxfactory.gen_alpha(), RAM, VCPUS, ZERO_DISK_SIZE)
+    wait_for(provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
+    zero_disk_flavor = provider.appliance.collections.cloud_flavors.instantiate(
+        api_flv.name, provider)
+
+    yield zero_disk_flavor
+
+    if zero_disk_flavor.exists:
+        zero_disk_flavor.delete(wait=False)
+
+
+@pytest.fixture(scope='function')
+def new_instance(provider, zero_disk_flavor):
+    flavor_name = zero_disk_flavor.name
+    prov_data = provider.data['provisioning']
+    prov_form_data = {
+        'request': {'email': fauxfactory.gen_email(),
+                    'first_name': fauxfactory.gen_alpha(),
+                    'last_name': fauxfactory.gen_alpha()},
+        'catalog': {'num_vms': '1',
+                    'vm_name': fauxfactory.gen_alpha()},
+        'environment': {'cloud_network': prov_data['cloud_network'],
+                        'cloud_tenant': prov_data['cloud_tenant']},
+        'properties': {'instance_type': partial_match(flavor_name)},
+    }
+
+    instance_name = prov_form_data['catalog']['vm_name']
+
+    try:
+        instance = provider.appliance.collections.cloud_instances.create(
+            instance_name,
+            provider,
+            prov_form_data, find_in_cfme=True
+        )
+
+    except KeyError:
+        # some yaml value wasn't found
+        pytest.skip('Unable to find an image map in provider "{}" provisioning data: {}'
+                    .format(provider, prov_data))
+
+    yield instance
+    try:
+        instance.cleanup_on_provider()
+    except Exception:
+        pass
+
+
+def test_create_instance_with_zero_disk_flavor(new_instance, soft_assert):
+    view = navigate_to(new_instance, 'Details')
+    prov_data = new_instance.provider.data['provisioning']
+    power_state = view.entities.summary('Power Management').get_text_of('Power State')
+    assert power_state == OpenStackInstance.STATE_ON
+
+    vm_tmplt = view.entities.summary('Relationships').get_text_of('VM Template')
+    soft_assert(vm_tmplt == prov_data['image']['name'])
+
+    flavors = [f.name for f in new_instance.provider.mgmt.api.flavors.list()]
+    flavor = view.entities.summary('Relationships').get_text_of('Flavor')
+    soft_assert(flavor in flavors)
+
+    # Assert other relationships in a loop
+    props = [('Availability Zone', 'availability_zone'),
```

Comment:
**Optional**

Widgetastic has a util function you can import to do this for you - just define the UI string (p = 'Availability Zone') and then use `widgetastic.utils.attributize_string(p)` to get `availability_zone`.

This means you don't have to define tuples and have `get_text_of(p[0])` and `prov_data[p[1]]`, you can just define strings and have `get_text_of(p)` and `prov_data.get(attributize_string(p))`

Reply:
https://github.com/RedHatQE/widgetastic.core/blob/master/src/widgetastic/utils.py#L493

Reply:
This wouldn't work for Virtual Private Cloud though, so, ignore me :)